vmm/src/device_manager.rs

```diff
@@ -423,20 +423,29 @@ impl DeviceManager {
             vm_fd: vm_info.vm_fd.clone(),
         });
 
-        let ioapic = DeviceManager::add_ioapic(vm_info, &address_manager)?;
-
         // Create a shared list of GSI that can be shared through all PCI
         // devices. This way, we can maintain the full list of used GSI,
         // preventing one device from overriding interrupts setting from
         // another one.
         let kvm_gsi_msi_routes: Arc<Mutex<HashMap<u32, KvmRoutingEntry>>> =
             Arc::new(Mutex::new(HashMap::new()));
 
+        let ioapic_interrupt_manager: Arc<dyn InterruptManager> =
+            Arc::new(KvmInterruptManager::new(
+                address_manager.allocator.clone(),
+                vm_info.vm_fd.clone(),
+                kvm_gsi_msi_routes.clone(),
+                None,
+            ));
+
```

Comment:
What about adding a `set_ioapic()` setter to `KvmInterruptManager` and having one single `InterruptManager` instead?

Reply:
I initially did that, but then ran into some issues. By doing that, this means the `KvmInterruptManager` must be mutable, which would change the entire codebase as we would need to pass and Arc<Mutex<dyn InterruptManager>> down to Ioapic and down to all VFIO and virtio-pci devices. I thought that would be too much, too invasive.

Reply:
I agree. And I'd rather have an immutable interrupt manager anyways.

The thing that bothers me here is the fact that we implicitly know that both interrupt managers are equivalent because we're `Arc::clone()`'ing the routes and allocator between them. Could we:

1. Add some explicit comments about this when creating both interrupt managers
2. Explictly calling `Arc::clone()` on the routes and allocator to be clear about the fact that those are going to be shared references (I know, this is picky...)

Reply:
Yes you're right there are some assumptions here, and adding some comments will definitely help.
Let me address 1 and 2 :)

Reply:
Done!